lib/internal/util/comparisons.js

```diff
@@ -237,9 +238,15 @@ function innerDeepEqual(val1, val2, strict, memos) {
     // is otherwise identical.
     if ((!isNativeError(val2) && !(val2 instanceof Error)) ||
         val1.message !== val2.message ||
-        val1.name !== val2.name) {
+        val1.name !== val2.name ||
+        !innerDeepEqual(val1.cause, val2.cause, strict, memos)) {
```

Comment:
This is definitely the right check as long the property is non-enumerable. If it is enumerable, we'd do the check twice and this could be costly and a lot of errors are created in the wild with a cause property set as an enumerable one.

In addition, we should also verify that the properties have the same enumerability. We don't yet do that for `name` and `message`, but that would be beneficial.

```js
const cause1Enumerable = ObjectPrototypePropertyIsEnumerable(val1, 'cause');
if (cause1Enumerable !== ObjectPrototypePropertyIsEnumerable(val2, 'cause') ||
    !cause1Enumerable &&
    !innerDeepEqual(val1.cause, val2.cause, strict, memos)) {
    return false;
}
```